doc/ipython-notebooks/link_checker.py

```diff
@@ -0,0 +1,54 @@
+import json
+import re
+import urllib.request, urllib.error
+import sys
+#requirement is python >= 3.5
+
+def check_python():
+    if (sys.version_info < (3,5)):
+        raise Exception("Python >= 3.5 required")
+
+def Find(cell_contents):
+    text = cell_contents[0]
+    urls = re.findall('http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', text)
+    return urls   
+
+def link_finder(file_name):
+    links = []
+    f = open(file_name)
+    s = f.read()
+    input_file = json.loads(s)
+    #convert given file to python dictionary fromat
+    cells_count = len(input_file['cells'])
+    for i in range(cells_count):
+        #Assuming all links are in markdown cells
+        if(input_file['cells'][i]['cell_type'] == 'markdown'):
+            urls_in_current_cell = Find(input_file['cells'][i]['source'])
+            links.extend(urls_in_current_cell)
+    return links       
+
+def link_checker(link_address):
+    try:
+        urllib.request.urlopen(link_address)
+    except urllib.error.HTTPError as e1:
+        # Return code error (e.g. 404, 501, ...)
+        return e1.code
+    except urllib.error.URLError as e2:
+        # Not an HTTP-specific error (e.g. connection refused)
+        return(e2.reason)
+    else:
+        return(200)
+    
+
+if __name__ == '__main__':
+    check_python()
+    # enter name of file
+    file_name = sys.argv[1]
+
+    links = link_finder(file_name)
+    number_of_links = len(links)
+    
+    for i in range(number_of_links):
```

Comment:
Can this not just be for link in links? No need for indexing or?

Reply:
Yes, I have tested this locally.It gives the following output for classification.ipynb 
under the directory shogun/doc/ipython-notebooks/classification
(it is working)
There is no need of indexing.I am removing them
![Screenshot 2020-03-14 at 2 14 27 PM](https://user-images.githubusercontent.com/41142068/76678467-457add80-65fe-11ea-9d65-99840a576545.png)
 :).